File: Assets/ToLua/Lua/user/ui/player/model/pet_data.lua
local t = {}
t.data = {}
local item = {}
item.__index = item

function item.New(table)
	local o = {}
	setmetatable(o,item)
	o.id = tostring(table.id)
	o.model = table.model
	o.scale = table.scale
	o.speed = tonumber(table.speed)
	o.home_rotation = table.home_rotation
	o.rotation = table.rotation
	o.stay_animation = table.stay_animation
	o.home_Offset = table.home_Offset
	o.Offset = table.Offset
	o.head_icon = table.head_icon
	return o
end

function t.GetDataById(id)
	return t.data[id]
end

local function Start()
	local origin = dofile('pet')
	origin.ForEach(function(id,table)
		local newItem = item.New(table)
		t.data[id] = newItem
	end)
end

Start()
return t

File: Assets/Code/Logic/Pet/Model/PetData.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using Common.Util;
namespace Logic.Pet.Model
{
    public class PetData
    {
        private static Dictionary<int, PetData> _petDataDictionary;

        public static Dictionary<int, PetData> GetPetDatas()
        {
            if (_petDataDictionary == null)
            {
                _petDataDictionary = CSVUtil.Parse<int, PetData>("config/csv/pet", "id");
            }
            return _petDataDictionary;
        }


        public static PetData GetPetDataByID(int petId)
        {
            if (_petDataDictionary == null)
                GetPetDatas();
            PetData petData = null;
            if (_petDataDictionary.ContainsKey(petId) && _petDataDictionary[petId] != null)
            {
                petData = _petDataDictionary[petId];
            }
            return petData;
        }

        [CSVElement("id")]
        public int id;

        //public string[] headIcons;
        //[CSVElement("headIcon")]
        //public string headIconStr
        //{
        //    set
        //    {
        //        headIcons = value.ToArray(CSVUtil.SYMBOL_SEMICOLON);
        //    }
        //}

        [CSVElement("model")]
        public string modelName;

        public Vector3 scale;
        [CSVElement("scale")]
        public string scaleString
        {
            set
            {
                scale = value.ToVector3(CSVUtil.SYMBOL_SEMICOLON);
            }
        }

        [CSVElement("speed")]
        public float speed;

        public Vector3 homeRotation;
        [CSVElement("home_rotation")]
        public string homeRotationString
        {
            set
            {
                homeRotation = value.ToVector3(CSVUtil.SYMBOL_SEMICOLON);
            }
        }

        public Vector3 rotation;
        [CSVElement("rotation")]
        public string rotationString
        {
            set
            {
                rotation = value.ToVector3(CSVUtil.SYMBOL_SEMICOLON);
            }
        }

        [CSVElement("stay_animation")]
        public string animation;

        public Vector3 homeOffset;
        [CSVElement("home_Offset")]
        public string homeOffsetString
        {
            set
            {
                homeOffset = value.ToVector3(CSVUtil.SYMBOL_SEMICOLON);
            }
        }

        public Vector3 offset;
        [CSVElement("Offset")]
        public string offsetString
        {
            set
            {
                offset = value.ToVector3(CSVUtil.SYMBOL_SEMICOLON);
            }
        }
		[CSVElement("head_icon")]
		public string head_icon;
    }
}